jax/experimental/jax2tf/tests/shape_poly_test.py

```diff
@@ -331,66 +332,258 @@ def test_stride_shape(self):
       core.stride_shape((a, 20), (1, 3), (2, 2))
 
 
-class ShapePolyTest(tf_test_util.JaxToTfTestCase):
+class PolyHarness(Harness):
+  """Tests a function with shape polymorphism.
+
+  Converts `fun` with shape polymorphism, creates a tf.ConcreteFunction
+  given `input_signature` and checks the inferred output shapes to match
+  `expected_output_shapes`, then checks that the JAX and the TF functions
+  produce the same results.
+
+  The following functionality is inherited from Harness:
+    * group_name, name: for naming the harness
+    * arg_descriptors: descriptors to create the arguments, works only for
+        functions that take simple (not pytree) arguments. If `skip_jax_run`
+        the JAX and TF functions are not executed.
+    * fun: a JAX function to convert with shape polymorphism.
+        If all `arg_descriptors` are just RandArg, then this function is used
+        directly. Otherwise, it is first partially applied to the StaticArg.
+
+
+  The `polymorphic_shapes` and `input_signature` for `jax2tf.convert` can be
+  specified directly or by way of the simpler `poly_axes`.
+  The `poly_axes` must correspond to the
+  non-static arguments, and for each one it must specify which axes are:
+  None, or an int (for the index of the polymorphic axis), or a tuple of ints
+  (for multiple polymorphic axes). For each argument, we use its `poly_axes`
+  entry to generate the polymorphic_shapes specification, creating dimension
+  variables `b0`, `b1, ..., for each of its polymorphic axes. This means that
+  separate arguments will share the same dimension variable names, in the order
+  in which the axes are listed in poly_axes. We also generate the input_signature
+  from `poly_axes`.
+
+  `check_result` specifies if we want to check that the result of the shape
+  polymorphic conversion produces the same result and the JAX function.
+
+  `expect_error` is a pair of an Exception type and a regular expression to
+  match the expected exception string.
+  """
+  def __init__(self,
+               group_name: str, name: str,
+               func: Callable,
+               *,
+               arg_descriptors: Sequence[primitive_harness.ArgDescriptor] = (),
+               poly_axes: Optional[Sequence[Optional[Union[int, Sequence[int]]]]] = None,
+               polymorphic_shapes: Optional[Sequence[Any]] = None,
+               input_signature: Optional[Sequence[tf.TensorSpec]] = None,
+               expected_output_signature: Optional[tf.TensorSpec] = None,
+               check_result=True,
+               skip_jax_run=False,
+               tol=None,
+               enable_xla=True,
+               expect_error=(None, None)):
+    super().__init__(group_name, name, func, arg_descriptors,
```

Comment:
Seems you call this `fun` in the description and in superclass `Harness`.

```suggestion
               fun: Callable,
               *,
               arg_descriptors: Sequence[primitive_harness.ArgDescriptor] = (),
               poly_axes: Optional[Sequence[Optional[Union[int, Sequence[int]]]]] = None,
               polymorphic_shapes: Optional[Sequence[Any]] = None,
               input_signature: Optional[Sequence[tf.TensorSpec]] = None,
               expected_output_signature: Optional[tf.TensorSpec] = None,
               check_result=True,
               skip_jax_run=False,
               tol=None,
               enable_xla=True,
               expect_error=(None, None)):
    super().__init__(group_name, name, fun, arg_descriptors,
```